src/m_who.c

```diff
@@ -254,21 +254,18 @@ int build_searchopts(aClient *sptr, int parc, char *parv[])
 			 sptr->name, "WHO", "who");
 	      return 0;
 	  }
-	  if(*parv[args] == '@' || *parv[args] == '+')
+	  if (*parv[args] == '@' || *parv[args] == '%' || *parv[args] == '+')
 	  {
 	      char *cname = parv[args] + 1;
 
-              if(*parv[args] == '@')
-	      {
-		  wsopts.channelflags = CHFL_CHANOP;
-		  if(*cname == '+')
-		  {
-		      wsopts.channelflags |= CHFL_VOICE;
-		      cname++;
-		  }
-	      }
-	      else
-		  wsopts.channelflags = CHFL_VOICE;
+		  if (*cname == '@')
```

Comment:
I think this could use some revisiting. We have to allow a who +c @%+#chan scenario as well. Perhaps it makes sense to check what param[args] is, then iterate through that to check if % was sent then else {} it with the VOICE flag. Thoughts?